src/routes/post-article.ts

```diff
@@ -0,0 +1,22 @@
+import { constants } from 'http2';
+import { Context, Middleware, Next } from 'koa';
+import { JsonLdObj } from 'jsonld/jsonld-spec';
+import uniqueString from 'unique-string';
+import createHttpError from 'http-errors';
+import Articles from '../articles';
+
+export default (articles: Articles): Middleware => (
+  async ({ request, response }: Context, next: Next): Promise<void> => {
+    const article = request.body as JsonLdObj;
+    if ('@id' in article) {
+      throw new createHttpError.Forbidden('Article ID is not permitted');
```

Comment:
Is there value is passing the id into the message for logging?

Perhaps: `Article ID is not permitted ('_:1' given)`

Reply:
Yeah, maybe even `Article IDs must not be set ('_:1' was given)`.